manageiq-operator/README.md

```diff
@@ -2,68 +2,125 @@
 ```
 
-2. Update the operator deployment with the new image:
+If it is not already available it can be deployed with command:
 
 ```bash
-$ sed -i 's|docker.io/manageiq/manageiq-operator:latest|docker.io/example/manageiq-operator:latest|g' deploy/operator.yaml
+$ oc create -f deploy/crds/manageiq.org_manageiqs_crd.yaml
 ```
 
-3. Push the new image to the registry:
+### Step 3. Run The Operator
+
+There are three different ways the operator can be run.
+
+
++ #### Option 1: Run the latest ManageIQ image from the registry
+
+  The default in the operator.yaml is for the latest manageiq-operator image.
+  So no change is required. Simply create the operator.
+
+  ```bash
+  $ oc create -f deploy/operator.yaml
+  ```
+
++ #### Option 2: Run your own custom ManageIQ image inside the Cluster
+
+  1 - Build your operator image:
+
+    ```bash
+    $ operator-sdk build docker.io/<your location>/manageiq-operator:latest
+    ```
+
+  2 - Update the operator deployment yaml file with your custome image:
+
+    ```bash
+    $ sed -i 's|docker.io/manageiq/manageiq-operator:latest|docker.io/<your location>/manageiq-operator:latest|g' deploy/operator.yaml
+    ```
+
+  3 - Push your new custom image to the registry:
+
+    ```bash
+    $ docker push docker.io/<your location>/manageiq-operator:latest
+    ```
+    
+  4 - Run your custom image from the registry:
+
+    ```bash
+    $ oc create -f deploy/operator.yaml
+    ```
+
++ #### Option 3: Run locally (on your local laptop/computer, outside of the cluster)
+
+  ```bash
+  $ operator-sdk run --local --namespace=<your namespace>
+  ```
+
+### Step 4. Perform any optional custom configurations
+
+*see the OpenID-Connect example below*
+
+### Step 5. Run ManageIQ by creating the Custom Resource (CR)
 
 ```bash
-$ docker push docker.io/example/manageiq-operator:latest
+$ oc create -f deploy/crds/manageiq.org_v1alpha1_manageiq_cr.yaml
 ```
 
-### Setup RBAC and deploy the operator
+# Further Notes:
 
-```bash 
-$ oc create -f deploy/role.yaml
-$ oc create -f deploy/role_binding.yaml
-$ oc create -f deploy/service_account.yaml
-$ oc create -f deploy/operator.yaml
-```
+## Customizing the installation
+
+ManageIQ can be run with an external Postgres or messaging server.  To do this, create the required OpenShift secret(s) with the correct parameters using the template(s) for [Postgres](templates/app/postgresql-secrets.yaml) or [messaging](/templates/app/kafka-secrets.yaml) and provide those secret names as `databaseSecret` and/or `kafkaSecret` in `manageiq.org_v1alpha1_manageiq_cr.yaml`.
 
-### Create the CR to deploy ManageIQ
+If you want to use a custom TLS certificate, one can be created with:
 
 ```bash
-$ oc create -f deploy/crds/manageiq.org_v1alpha1_manageiq_cr.yaml
+oc create secret tls tls-secret --cert=tls.crt --key=tls.key` and setting the secret name as `tlsSecret` in `manageiq.org_v1alpha1_manageiq_cr.yaml`.
 ```
 
-**ManageIQ Instance Example**
+## Creating an Operator Bundle
```

Comment:
Can this be above or below the `Customizing the installation` section?  It's in the middle right now and it's not related to customizing the installation.